src/TiledArray/math/linalg/forward.h

```diff
@@ -0,0 +1,49 @@
+/*
+ *  This file is a part of TiledArray.
+ *  Copyright (C) 2020 Virginia Tech
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *  David Williams-Young
+ *  Computational Research Division, Lawrence Berkeley National Laboratory
+ *
+ *  svd.h
+ *  Created:    12 June, 2020
+ *
+ */
+#ifndef TILEDARRAY_MATH_LINALG_FORWARD_H__INCLUDED
+#define TILEDARRAY_MATH_LINALG_FORWARD_H__INCLUDED
+
+#include <TiledArray/config.h>
+#include <type_traits>
+
+namespace TiledArray::math::linalg {
+
+enum TransposeFlag { NoTranspose, Transpose, ConjTranspose };
+
+enum class SVDReturn {
```

Comment:
I originally had these as strong types to allow for the possibility of  typed template specialization for particular combinations (the performant alg choices for thin return / value only may be slightly different than full return, etc). I'm fine keeping the change, but we _**may**_ find use cases for the strong types later on if we look at more advanced dispatch